frontend/src/components/TodoListItem.tsx

```diff
@@ -1,20 +1,42 @@
+import { ChangeEvent, useState } from "react";
 import { Todo } from "../models/todo.model";
 import { IconDelete } from "./Icons/IconDelete";
 
 interface TodoListItemProps {
   todo: Todo;
-  deleteTodo: (id: string) => void;
+  onDeleteTodo: (id: string) => void;
+  onChangeTodo: (id: string, check: boolean) => void;
 }
-export default function TodoListItem({ todo, deleteTodo }: TodoListItemProps) {
+export default function TodoListItem({
+  todo,
+  onChangeTodo,
+  onDeleteTodo,
+}: TodoListItemProps) {
+  const [check, setCheck] = useState(!todo.isDone);
+  const valueChangeHandler = (event: ChangeEvent<HTMLInputElement>) => {
+    onChangeTodo(todo.id, event.target.checked);
+    setCheck(!check);
```

Comment:
Again, this is the wrong way to update state if it is based on previous state. Furthermore, you do not need to store state for the check here in this component. With this approach, the array of Todos remains the same in the App component.

Remove the state here, and update the Todos array in the App component.

Reply:
@Ejazhussainbaltistani You marked this conversation as resolved, but you did not make the changes